### ISSUE TITLE
test task cannot be excluded with setExcludedTaskNames

### ISSUE THREAD

**Issue by matti-1** [gradle/gradle#16756]
### Expected Behavior
`test` task should be excluded with `getStartParameter().setExcludedTaskNames(["test"])`.

### Current Behavior
Test task cannot be excluded when creating a `GradleBuild` task which includes a `build` task:
```
plugins { id "java-library" }

task("exampleTask", type: GradleBuild) {  
  tasks = ["build"]
  getStartParameter().setExcludedTaskNames(["test"])
}
```

By running the build.gradle above the test task is still included:
```
$ gradle exampleTask

> Task :test-project:compileJava NO-SOURCE
> Task :test-project:processResources NO-SOURCE
> Task :test-project:classes UP-TO-DATE
> Task :test-project:jar
> Task :test-project:assemble
> Task :test-project:compileTestJava NO-SOURCE
> Task :test-project:processTestResources NO-SOURCE
> Task :test-project:testClasses UP-TO-DATE
> Task :test-project:test NO-SOURCE
> Task :test-project:check UP-TO-DATE
> Task :test-project:build
> Task :exampleTask

BUILD SUCCESSFUL in 14s
1 actionable task: 1 executed
```

### Context
After updating to Gradle `6.8.3` the test task cannot be excluded. This still worked with `6.7.1` version. I tested this with versions `6.8` and `7.0-rc-2` and problem occurs also with these versions.

With version `6.7.1` the test task is excluded:
```
$ gradle exampleTask

> Task :test-project:compileJava NO-SOURCE
> Task :test-project:processResources NO-SOURCE
> Task :test-project:classes UP-TO-DATE
> Task :test-project:jar
> Task :test-project:assemble
> Task :test-project:check
> Task :test-project:build
> Task :exampleTask

BUILD SUCCESSFUL in 13s
1 actionable task: 1 executed
```

In some cases this has been useful feature to exclude test task to get a build done quickly.

### Steps to Reproduce 

To reproduce, run exampleTask with following build.gradle:
```
plugins { id "java-library" }

task("exampleTask", type: GradleBuild) {  
  tasks = ["build"]
  getStartParameter().setExcludedTaskNames(["test"])
}
```
No actual source or test files are required for reproducing this.

**Comment by oesolutions** [comment]
I'm unsure about this `isRootBuild` check:
https://github.com/gradle/gradle/blob/e72eb95462cb920937e6eef244144399877532fa/subprojects/core/src/main/java/org/gradle/execution/CompositeAwareTaskSelector.java#L54
If the filter path is not part of an included build then shouldn't it always defer to the default task selector?
When debugging a `GradleBuild` task I see that `isRootBuild` returns false because it has a parent (the build containing the `GradleBuild` task). If I force the return value of `isRootBuild` to be true, then the tasks are correctly filtered.
Perhaps the issue isn't the root build check, but rather that the build should not have a parent set? I'm unsure what the appropriate action is, but hopefully this helps someone with more overall knowledge.

**Comment by rk-puvvada** [comment]
Would there be a temporary workaround for this issue to those who are using 6.8.x or 6.9 versions of Gradle distribution? It was a regressions after upgrading the gradle distribution version to 6.8.3 from 6.7 version.

**Comment by eskatos** [comment]
This was fixed in Gradle 7.6
